tools/src/main/java/org/neo4j/tools/dump/InconsistencyReportReader.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.tools.dump;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+
+/**
+ * Reads CC inconsistency reports. Example of entry:
+ *
+ * <pre>
+ * ERROR: The referenced relationship record is not in use.
+ *     Node[3496089,used=true,rel=14833798,prop=13305361,labels=Inline(0x1000000006:[6]),light,secondaryUnitId=-1]
+ *     Inconsistent with: Relationship[14833798,used=false,source=0,target=0,type=0,sPrev=0,sNext=0,tPrev=0,tNext=0,prop=0,secondaryUnitId=-1,!sFirst,!tFirst]
+ * </pre>
+ */
+public class InconsistencyReportReader
+{
+    private final Inconsistencies inconsistencies;
+
+    public interface Inconsistencies
+    {
+        void node( long id );
+
+        void relationship( long id );
+
+        void property( long id );
+
+        void relationshipGroup( long id );
+
+        // TODO: incomplete list/arguments, driven by actual need a.t.m.
+    }
+
+    public InconsistencyReportReader( Inconsistencies inconsistencies )
+    {
+        this.inconsistencies = inconsistencies;
+    }
+
+    public void read( File file ) throws IOException
+    {
+        try ( Reader reader = new FileReader( file ) )
+        {
+            read( reader );
+        }
+    }
+
+    public void read( Reader reader ) throws IOException
+    {
+        int state = 0; // 0:inconsistency description, 1:entity, 2:inconsistent with
+        BufferedReader bufferedReader = new BufferedReader( reader );
+        String line;
+        while ( (line = bufferedReader.readLine()) != null )
+        {
+            line = line.trim();
+            if ( state == 0 )
+            {
+                if ( !line.contains( "ERROR" ) && !line.contains( "WARNING" ) )
+                {
+                    break;
+                }
+            }
+            else if ( state == 1 )
+            {
+                String entityType = entityType( line );
```

Comment:
entity type load and propagation (line 84- 89) can be extracted since exactly the same code used in next brach